app/connectors/ReferenceDataConnector.scala

```diff
@@ -38,102 +38,110 @@ import scala.reflect.ClassTag
+    val queryParams = Seq("data.id" -> customsOfficeId)
+    val url         = url"${config.customsReferenceDataUrl}/lists/CustomsOffices?$queryParams"
+    getOrElseUpdate[CustomsOffice](url)
   }
 
-  def getCountries(
-    queryParams: QueryParams*
-  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[Country]] = {
-    val url = url"${config.customsReferenceDataUrl}/lists/CountryCodesFullList?$queryParams"
-    get[Country](url)
+  def getCountry(
+    code: String
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Response[Country]] = {
+    val queryParams = Seq("data.code" -> code)
+    val url         = url"${config.customsReferenceDataUrl}/lists/CountryCodesFullList?$queryParams"
+    getOrElseUpdate[Country](url)
   }
 
-  def getQualifierOfIdentifications(
-    queryParams: QueryParams*
-  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[QualifierOfIdentification]] = {
-    val url = url"${config.customsReferenceDataUrl}/lists/QualifierOfTheIdentification?$queryParams"
-    get[QualifierOfIdentification](url)
+  def getQualifierOfIdentification(
+    qualifier: String
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Response[QualifierOfIdentification]] = {
+    val queryParams = Seq("data.qualifier" -> qualifier)
+    val url         = url"${config.customsReferenceDataUrl}/lists/QualifierOfTheIdentification?$queryParams"
+    getOrElseUpdate[QualifierOfIdentification](url)
   }
 
-  def getIdentificationTypes(
-    queryParams: QueryParams*
-  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[IdentificationType]] = {
-    val url = url"${config.customsReferenceDataUrl}/lists/TypeOfIdentificationOfMeansOfTransport?$queryParams"
-    get[IdentificationType](url)
+  def getIdentificationType(
+    `type`: String
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Response[IdentificationType]] = {
+    val queryParams = Seq("data.type" -> `type`)
+    val url         = url"${config.customsReferenceDataUrl}/lists/TypeOfIdentificationOfMeansOfTransport?$queryParams"
+    getOrElseUpdate[IdentificationType](url)
   }
 
-  def getNationalities(
-    queryParams: QueryParams*
-  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[Nationality]] = {
-    val url = url"${config.customsReferenceDataUrl}/lists/Nationality?$queryParams"
-    get[Nationality](url)
+  def getNationality(
+    code: String
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Response[Nationality]] = {
+    val queryParams = Seq("data.code" -> code)
+    val url         = url"${config.customsReferenceDataUrl}/lists/Nationality?$queryParams"
+    getOrElseUpdate[Nationality](url)
   }
 
   def getIncidentCode(
     queryParams: QueryParams*
-  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[IncidentCode] = {
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Response[IncidentCode]] = {
```

Comment:
I still think `QueryParams*` is a poor input type for an expected return type of `Response[T]` (i.e. an individual item from the code list).

If we know the purpose of the method is to find one item from the code list, then the input parameter should be some unique field (an id for example).

Same applies for `getIncidentCode`, `getControlType`, `getRequestedDocumentType`, `getFunctionalError`, and `getInvalidGuaranteeReason`.